### PR TITLE
Refactor project structure

### PR DESCRIPTION
**Pull request by mmonzo-eb**
We created a folder called 'api', where we put the files related to the api behaviour. We also implemented a mixin to return a set of model objects to the context of the view.

### CODE REVIEW

File: search_events_app/models/country.py [hunk 1]
```diff
@@ -27,3 +29,16 @@ def __str__(self):
...
+
+    @classmethod
+    def get_context(cls):
+        countries = cls.objects.all()
+
+        return {
+            'countries': [
+                {
+                    'alpha2Code': country.alpha_2_code,
+                    'name': country.name
```
Comment (walterbucolo-eb) [inline search_events_app/models/country.py:41]: Trailing comma

File: search_events_app/models/language.py [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+    
+    @classmethod
+    def get_context(cls):
+        languages = cls.objects.order_by('name')
+
+        return {
+            'languages': [
+                {
+                    'code': language.code,
+                    'name': language.name
```
Comment (walterbucolo-eb) [inline search_events_app/models/language.py]: trailing comma

File: search_events_app/tests/models/test_country.py [hunk 3]
```diff
@@ -16,3 +21,33 @@ def test_country_basic_info(self):
 
     def test_country_str(self):
         self.assertEqual(self.country.__str__(), 'Argentina')
+
+    @patch.object(
+		Country,
+		'objects'
+	)
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_country.py:28]: Paths could be in one line for more readability

Comment (walterbucolo-eb) [reply search_events_app/tests/models/test_country.py:28]: In this case, for example, just put
`@patch.object(Country,	'objects')`

File: search_events_app/tests/models/test_country.py [hunk 4]
```diff
@@ -16,3 +21,33 @@ def test_country_basic_info(self):
...
+    def test_get_context(self, mock_objects):
+        countries = [
+			Country(label='Argentina', code='AR', eventbrite_id='1234'),
+			Country(label='Spain', code='ES', eventbrite_id='4567')
+		]
+        
+        mock_objects.all = MagicMock(return_value=countries)
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_country.py:35]: Are you patching `countries = cls.objects.all()` ? Are you sure that this call return a list? Maybe return an object type _queryset_. So, you need to return an object with fake data but with the same type.

File: search_events_app/tests/models/test_language.py [hunk 5]
```diff
@@ -14,3 +19,33 @@ def test_language_basic_info(self):
 
     def test_language_str(self):
         self.assertEqual(self.language.__str__(), 'English')
+
+    @patch.object(
+		Language,
+		'objects'
+	)
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_language.py:26]: Same as above

File: search_events_app/tests/models/test_language.py [hunk 6]
```diff
@@ -14,3 +19,33 @@ def test_language_basic_info(self):
...
+    def test_get_context(self, mock_objects):
+        languages = [
+			Language(name='Spanish', code='es'),
+			Language(name='German', code='de')
+		]
+        
+        mock_objects.order_by = MagicMock(return_value=languages)
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_language.py:33]: Same as above

File: search_events_app/tests/services/api/test_api_service.py [hunk 7]
```diff
@@ -98,10 +97,10 @@ def setUp(self):
         'format_body'
     )
     @patch(
-        'search_events_app.services.api_service.requests.post'
+        'search_events_app.services.api.api_service.requests.post'
```
Comment (walterbucolo-eb) [inline search_events_app/tests/services/api/test_api_service.py:100]: I think is not needed this newline

File: search_events_app/tests/services/api/test_api_service.py [hunk 8]
```diff
@@ -98,10 +97,10 @@ def setUp(self):
...
     )
     @patch(
-        'search_events_app.services.api_service.process_events'
+        'search_events_app.services.api.api_service.process_events'
```
Comment (walterbucolo-eb) [inline search_events_app/tests/services/api/test_api_service.py:103]: same as above

File: search_events_app/views.py [hunk 9]
```diff
@@ -20,21 +20,11 @@ def get_queryset(self):
...
-        ]
-
-        context['languages'] = [
-            {
-                'code': language.code,
-                'name': language.name
-            } for language in languages
-        ]
+
+        classes = [Country, Language]
```
Comment (walterbucolo-eb) [inline search_events_app/views.py:25]: I not sure whats mean this line... could you explain me?

File: search_events_app/tests/models/test_country.py (hunk 3 above)
Comment (mmonzo-eb) [reply search_events_app/tests/models/test_country.py:28]: Resolved

File: search_events_app/tests/services/api/test_api_service.py (hunk 7 above)
Comment (mmonzo-eb) [reply search_events_app/tests/services/api/test_api_service.py:100]: I don't understand what you mean. We put the files related to the api in a folder called 'api'

File: search_events_app/tests/models/test_country.py [hunk 10]
```diff
@@ -16,3 +21,33 @@ def test_country_basic_info(self):
...
+
+        expected_result = {
+            'countries': [
+                {
+                    'alpha2Code': 'AR',
+                    'name': 'Argentina'
+                },
+                {
+                    'alpha2Code': 'ES',
+                    'name': 'Spain'
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_country.py:45]: Trailing comma

File: search_events_app/views.py (hunk 9 above)
Comment (mmonzo-eb) [reply search_events_app/views.py:25]: We have created a class called GetContextMixin, which is inherited by Country and Language, and will be inherited by Feature in the future. So we add more classes to that 'classes' array to fill the context variable

File: search_events_app/tests/models/test_language.py [hunk 11]
```diff
@@ -6,11 +11,38 @@
...
         self.assertEqual(self.language.name, 'English')
-        self.assertEqual(self.language.code, 'EN')
+        self.assertEqual(self.language.code, 'en')
 
     def test_language_str(self):
         self.assertEqual(self.language.__str__(), 'English')
+
+    @patch.object(Language.objects, 'order_by')
+    def test_get_context(self, mock_objects):
+        languages = [
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_language.py:25]: Do you need to do the same changes here?

File: search_events_app/tests/models/test_language.py [hunk 12]
```diff
@@ -6,11 +11,38 @@
...
+        expected_result = {
+            'languages': [
+                {
+                    'code': 'es',
+                    'name': 'Spanish'
+                },
+                {
+                    'code': 'de',
+                    'name': 'German'
+                }
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_language.py:41]: trailing comma

File: search_events_app/tests/models/test_country.py [hunk 13]
```diff
@@ -1,18 +1,46 @@
...
+                }
+            ]
+        }
+
+        result = Country.get_context()
+
+        self.assertEqual(result, expected_result)
+
```
Comment (walterbucolo-eb) [inline search_events_app/tests/models/test_country.py:46]: newline

**Pull request merged**